### DISCUSSION TITLE
\eqref renders as (???)

### DISCUSSION THREAD

**Discussion by smalltimer** [remarkjs/remark#570] [Q&A] [answered]
Hi,

I'm trying to use gatsby-remark-mathjax-ssr to render equations and reference them in md files. So my markdown contains something like:

```
$$
x + y = z
\label{1} \tag{1}
$$
```

which is rendered very beautifully. But when I reference it later with, e.g., In equation `$\eqref{1}$` ... the rendered text reads `In equation (???) ...`. To me, it seems like no # anchors are generated at all.

Any ideas what might be going wrong here?

Cheers

**Comment by ChristianMurphy** [comment, marked as answer]
@smalltimer could you share a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of what you are seeing?
Ideally isolating the issue without gatsby, which has a lot of unrelated pieces.
https://codesandbox.io/s/remark-debug-ikwvx or https://codesandbox.io/s/remark-rehype-debug-4cz8v could be a good starting point

**Reply by ChauhanT** [reply]
Hi @ChristianMurphy , thank you for getting back to me so quickly. Really appreciate it. The error occurs [here](http://www.tusharchauhan.com/writing/2020/09/models-of-synaptic-conductance-ii).

The problem is that I am not very good with web programming and the website is just a hobby project where I tinker around with code (mainly going through gatsby tutorials...). I think isolating the issue from gatsby will be very hard for me. If you give me some tips on how I can do this, I will of course try my best to.

**Reply by wooorm** [reply]
One important thing: `gatsby-remark-mathjax-ssr` has more to do with gatsby than with remark and is not maintained by us. You might be better off asking on that specific project, or on gatsby.

**Reply by ChristianMurphy** [reply]
@smalltimer that's great that you're getting into tinkering with Gatsby and other software!
A pointer: respecting time goes both ways, asking another hobbyist/volunteer the equivalent of "please debug my Gatsby app for me" is a bit much :slightly_smiling_face:, plus not sharing the source makes it near impossible

> If you give me some tips on how I can do this, I will of course try my best to.

Gatsby's `gatsby-remark-mathjax-ssr` package is a wrapper around remark,
https://codesandbox.io/s/remark-rehype-debug-4cz8v renders markdown to html,
combining it with https://github.com/remarkjs/remark-math/tree/main/packages/remark-math and https://github.com/remarkjs/remark-math/tree/main/packages/rehype-mathjax would provide a more minimal example.

---

More generally, I suspect this is an issue with MathJax.
In this example https://astexplorer.net/#/gist/61bfd0f2b3291f92c1632b6849ca7146/a947843a4a972bc6ed54dab99fa474cf0a3e3fac
`remark` and `remark-math` extract the math block as expected, it then gets passed to a math library like MathJax or KaTeX to do the actual mathematics handling.

**Reply by ChauhanT** [reply]
@wooorm Thanks. I suspected as much. I have opened an issue at the `gatsby-plubin-mathjax-ssr `repo but thought I'd ask here and see if someone might have faced a similar problem.

@ChristianMurphy Fair dues, and thanks again for responding so quickly! I wanted to try over the weekend to resolve the issue and report here before responding. Thanks for the links. I'll try to tinker with the ssr wrapper and see if I can get it to work. I am new to the whole idea of sharing gists and working code etc. :D If I resolve this I'll report back for other users who might be running into similar issues. 

Cheers guys. Very fast responses as ever!
